components/ipfs/ipfs_utils.cc

```diff
@@ -425,4 +425,41 @@ std::string GetRegistryDomainFromIPNS(const GURL& url) {
       cid, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
 }
 
+absl::optional<GURL> TranslateToCurrentGatewayUrl(const GURL& url) {
+  if (!url.is_valid() || !url.SchemeIsHTTPOrHTTPS()) {
+    return absl::nullopt;
+  }
+
+  std::vector<std::string> host_parts = base::SplitStringUsingSubstr(
+      url.host(), ".", base::KEEP_WHITESPACE, base::SPLIT_WANT_ALL);
+
+  if (host_parts.size() > 2 && IsValidCID(host_parts.at(0)) &&
```

Comment:
having an example of what original url transforms to which final url would be handy